src/phes_base.cpp

```diff
@@ -217,35 +220,50 @@ GeographicCoordinate get_origin(double latitude, double longitude, int border){
 	return GeographicCoordinate_init(FLOOR(latitude)+1+(border/3600.0),FLOOR(longitude)-(border/3600.0));
 }
 
-ExistingReservoir get_existing_reservoir(string name) {
+ExistingReservoir get_existing_reservoir(string name, string filename) {
   ExistingReservoir to_return;
-  string filename = file_storage_location + "input/existing_reservoirs/" + existing_reservoirs_csv;
+  int i = 0;
+  if (filename.empty())
+    filename = file_storage_location + "input/existing_reservoirs/" + existing_reservoirs_csv;
   if(!file_exists(convert_string(filename))){
     cout << "File " << filename << " does not exist." << endl;
     throw 1;
   }
-  vector<ExistingReservoir> reservoirs = read_existing_reservoir_data(
-      convert_string(filename));
-
-  for (ExistingReservoir r : reservoirs)
-    if (r.identifier == name)
-      to_return = r;
 
-  int i = 0;
-  for (string s : read_names(convert_string(file_storage_location +
-                                            "input/existing_reservoirs/" +
-                                            existing_reservoirs_shp_names))) {
-    if (s == name)
-      break;
-    else
-      i++;
+  if (use_tiled_bluefield) {
+    DBFHandle DBF = DBFOpen(convert_string(filename), "rb");
+    int dbf_field = DBFGetFieldIndex(DBF, string("Vol_total").c_str());
+    int dbf_elevation_field = DBFGetFieldIndex(DBF, string("Elevation").c_str());
+    int dbf_name_field = DBFGetFieldIndex(DBF, string("Lake_name").c_str());
+    for (i = 0; i<DBFGetRecordCount(DBF); i++){
+      const char* s = DBFReadStringAttribute(DBF, i, dbf_name_field);
+      if(name==string(s)){
+        break;
+      }
+    }
+    if(i==DBFGetRecordCount(DBF)){
+      cout<<"Could not find reservoir with name " << name << " in " << filename << endl;
+      throw 1;
+    }
+    to_return = ExistingReservoir_init(name, 0, 0, DBFReadIntegerAttribute(DBF, i, dbf_elevation_field), DBFReadDoubleAttribute(DBF, i, dbf_field));
+    DBFClose(DBF);
+  } else {
+    vector<ExistingReservoir> reservoirs = read_existing_reservoir_data(convert_string(filename));
+
+    for (ExistingReservoir r : reservoirs)
+      if (r.identifier == name)
+        to_return = r;
```

Comment:
Might be worth having handling for case where "name" is not the identifier of any reservoir in the CSV - I think it would be relevant for single_existing since the name is pulled from search_config.name

Reply:
Fixed